src/components/NewKeyPairForm.tsx

```diff
@@ -0,0 +1,195 @@
+import styled from "styled-components";
+import { Keypair } from "stellar-sdk";
+import CopyToClipboard from "react-copy-to-clipboard";
+
+const WarningEl = styled.div`
+  background-color: #f3e5e5;
+  color: #681e1e;
+  padding: 20px;
+  margin-bottom: 20px;
+`;
+
+const TempButtonEl = styled.button`
+  margin-bottom: 20px;
+`;
+
+const KeyPairWrapperEl = styled.div`
+  margin-bottom: 20px;
+`;
+
+const ConfirmWrapperEl = styled.div`
+  margin-bottom: 20px;
+`;
+
+const TempErrorEl = styled.div`
+  color: #c00;
+  margin-bottom: 20px;
+`;
+
+interface KeyPairType {
+  publicKey: string;
+  secretKey: string;
+}
+
+interface NewKeyPairFormProps {
+  onClose?: () => void;
+}
+
+export const NewKeyPairForm = ({ onClose }: NewKeyPairFormProps) => {
+  const [acceptedWarning, setAcceptedWarning] = useState(false);
+  const [newKeyPair, setNewKeyPair] = useState<KeyPairType | undefined>();
+  const [keyPairCopyString, setKeyPairCopyString] = useState("");
+  const [isKeyPairCopied, setIsKeyPairCopied] = useState(false);
+  const [confirmSavedSecretKey, setConfirmSavedSecretKey] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const generateNewKeyPair = () => {
+    const keypair = Keypair.random();
+
+    setNewKeyPair({
+      publicKey: keypair.publicKey(),
+      secretKey: keypair.secret(),
+    });
+
+    // Spacing here is important for copied string
+    setKeyPairCopyString(`Public key:
+${keypair.publicKey()}
+Secret key:
+${keypair.secret()}`);
+  };
+
+  const handleContinue = () => {
+    setAcceptedWarning(true);
+    generateNewKeyPair();
+  };
+
+  const handleDone = () => {
+    if (!confirmSavedSecretKey) {
+      setErrorMessage(
+        "Please confirm that you have copied and saved your secret key",
+      );
+      return;
+    }
+
+    handleClose();
+  };
+
+  const handleClose = () => {
+    if (onClose) {
+      onClose();
+    }
+  };
+
+  const handleCopyKeys = () => {
+    setIsKeyPairCopied(true);
+  };
+
+  const toggleConfirmSavedSecretKey = () => {
+    setErrorMessage("");
+    setConfirmSavedSecretKey(!confirmSavedSecretKey);
+  };
+
+  return (
+    <>
+      {/* Show warning */}
+      {!acceptedWarning && (
+        <div>
+          <h2>Generate a new key pair</h2>
+
+          <WarningEl>
+            <h3>
+              ATTENTION: Copying and pasting your secret key is not recommended
+            </h3>
+
+            <ul>
+              <li>
+                By copying and pasting your secret key you are vulnerable to
+                different attacks and scams that can result in your secret key
+                being stolen.
+              </li>
+              <li>Only use this option if you're aware of the risks...</li>
+              <li>
+                Ideally use other authentication methods like a hardware wallet
+                or a browser
+              </li>
+              <li>...</li>
+            </ul>
+          </WarningEl>
```

Comment:
This warning doesn't seem to apply to this page? The user's not going to paste a key into this form.

Reply:
I'm assuming this is just a placeholder warning text for now in wireframes.